java/org/apache/coyote/http11/Http11Processor.java

```diff
@@ -915,8 +915,26 @@ protected final void prepareResponse() throws IOException {
                 headers.addValue(Constants.CONNECTION).setString(
                         Constants.CLOSE);
             }
-        } else if (!http11 && !getErrorState().isError()) {
-            headers.addValue(Constants.CONNECTION).setString(Constants.KEEPALIVE);
+        } else if (!getErrorState().isError()) {
+            if (!http11) {
+                headers.addValue(Constants.CONNECTION).setString(Constants.KEEPALIVE);
```

Comment:
Is this okay for, say, HTTP 0.9? I realize that this code was previously there.

Reply:
As far as I know, HTTP did never exist, did it? I simply have broken up the clause. I have no opinion here.

Reply:
HTTP/0.9 should not be a concern. For HTTP/0.9 requests, the method exits much earlier.